app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/viewmodel/MoviesDetailsViewModelFactory.kt

```diff
@@ -0,0 +1,13 @@
+package com.bashkevich.androidfundamentals.moviesdetails.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.bashkevich.androidfundamentals.data.JsonLoad
+
+@Suppress("UNCHECKED_CAST")
+class MoviesDetailsViewModelFactory : ViewModelProvider.Factory{
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T = when (modelClass) {
+        MoviesDetailsViewModel::class.java -> MoviesDetailsViewModel(JsonLoad())
```

Comment:
You can even pass movie id here.

Reply:
I don't think it is necessary. And I don't know how to pass it in the ViewModelFactory

Reply:
I would try it as an exercise. That will be helpful in the future.

Reply:
Maybe in the different PR